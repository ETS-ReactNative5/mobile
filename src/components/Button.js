import React from 'react'
import {
  Text,
  TouchableOpacity
} from 'react-native'
import PropTypes from 'prop-types';
import EStyleSheet from 'react-native-extended-stylesheet'
import { append } from 'ramda'

const Button = (props) => {
  let buttonStyle = ( props.buttonStyle ? [styles.button, styles[props.buttonStyle]] : styles.button )
  buttonStyle = (props.additionalStyles ? append(props.additionalStyles, [buttonStyle]) : buttonStyle)

  const textStyle = (props.additionalTextStyles ? append(props.additionalTextStyles, [styles.buttonText]) : styles.buttonText)

  return (
    <TouchableOpacity
      style={buttonStyle}
      onPress={props.handlePress}
      disabled={props.disabled}
      activeOpacity={0.5}>
      <Text style={textStyle}>
        {props.text}
      </Text>
    </TouchableOpacity>
  )
}

const styles = EStyleSheet.create({
  buttonText: {
    fontSize: 14,
    color: 'white',
    alignSelf: 'center',
    letterSpacing: 1.3,
    flex: 1,
  },
  button: {
    borderRadius: 4,
    padding: 10,
    flexDirection: 'row',
    backgroundColor: '$buttonColor',
    alignSelf: 'stretch',
    justifyContent: 'center',
    marginTop: 10
  },
  disabledButton: {
    backgroundColor: '$disabledButtonColor'
  },
  registerButton: {
    backgroundColor: '$registerButtonColor'
  },
  disabledRegisterButton: {
    backgroundColor: '$registerDisabledButtonColor'
  },
  navyButton: {
    backgroundColor: '$navy',
  },
});

Button.propTypes = {
  handlePress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  buttonStyle: PropTypes.string,
  additionalStyles: PropTypes.array,
  additionalTextStyles: PropTypes.array,
  text: PropTypes.string
}
Button.defaultProps = { disabled: false }

export default Button
