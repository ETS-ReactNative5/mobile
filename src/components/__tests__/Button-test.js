import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../Button'

it('renders button text correctly', () => {
  const tree = renderer.create(
    <Button text={'O Hai!'} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders TouchableOpacity disabled when passed in', () => {
  const tree = renderer.create(
    <Button disabled={true} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders styles as array when special style is passed in', () => {
  const tree = renderer.create(
    <Button buttonStyle={'hai'} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
