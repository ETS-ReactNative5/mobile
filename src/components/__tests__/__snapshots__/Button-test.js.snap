exports[`test renders TouchableOpacity disabled when passed in 1`] = `
<TouchableOpacity
  activeOpacity={0.5}
  disabled={true}
  onPress={undefined}
  style={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined} />
</TouchableOpacity>
`;

exports[`test renders button text correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.5}
  disabled={false}
  onPress={undefined}
  style={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}>
    O Hai!
  </Text>
</TouchableOpacity>
`;

exports[`test renders styles as array when special style is passed in 1`] = `
<TouchableOpacity
  activeOpacity={0.5}
  disabled={false}
  onPress={undefined}
  style={
    Array [
      undefined,
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined} />
</TouchableOpacity>
`;
